core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java

```diff
@@ -34,20 +34,23 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
 import static io.airlift.slice.SizeOf.sizeOf;
 import static io.trino.spi.StandardErrorCode.GENERIC_INSUFFICIENT_RESOURCES;
 import static io.trino.spi.type.BigintType.BIGINT;
 import static io.trino.type.TypeUtils.NULL_HASH_CODE;
 import static io.trino.util.HashCollisionsEstimator.estimateNumberOfHashCollisions;
 import static it.unimi.dsi.fastutil.HashCommon.arraySize;
 import static it.unimi.dsi.fastutil.HashCommon.murmurHash3;
+import static java.lang.Math.min;
 import static java.lang.Math.toIntExact;
 import static java.util.Objects.requireNonNull;
 
 public class BigintGroupByHash
         implements GroupByHash
 {
     private static final int INSTANCE_SIZE = ClassLayout.parseClass(BigintGroupByHash.class).instanceSize();
+    private static final int BATCH_SIZE = 1024;
```

Comment:
nit: this is bigger than expected groups for `io.trino.sql.planner.LocalExecutionPlanner.Visitor#visitTableWriter` and `io.trino.sql.planner.LocalExecutionPlanner.Visitor#visitTableFinish` (200)

Reply:
Expected number of groups is hardcoded as `10000` so I don't think that matters.